### PR TITLE
[DEV-1064] Add image retire/promote functionality

### PR DESCRIPTION
**Pull request by g0rgamesh**
- added function to retire old images
- added function to promote new images
- added switch for image build function as it was needed for standalone image promotion functionality

### CODE REVIEW

File: tasks/promote.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Ensure old images are retired
+  command: >
+            {{ os_images_venv }}/bin/openstack image set {{ promotion_name }} --name {{ promotion_name }}.{{ date_suffix }}
```
Comment (mnasiadka) [inline tasks/promote.yml]: either you do a multiline, or drop > ;-)

File: tasks/promote.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  loop: "{{ os_images_list }}"
+  when: item.rename_image | default(os_images_promote) | bool
+  environment: "{{ os_images_venv }}"
+
+- name: Ensure new images are promoted
+  command: >
+            {{ os_images_venv }}/bin/openstack image set {{ item.name }} --name {{ promotion_name }}
+  vars:
+    suffix: "{{ os_images_name_suffix }}"
+    suffix_length: "{{ os_images_name_suffix|length }}"
```
Comment (mnasiadka) [inline tasks/promote.yml]: What if this is empty?

Comment (g0rgamesh) [reply tasks/promote.yml]: Added a length check for this variable

File: tasks/promote.yml (hunk 1 above)
Comment (g0rgamesh) [reply tasks/promote.yml]: Done

File: tasks/promote.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+- name: Check if image suffix is provided
+  fail:
+    msg: "os_images_name_suffix is empty please provide it."
+  when: os_images_name_suffix is defined and os_images_name_suffix|length == 0
```
Comment (mnasiadka) [inline tasks/promote.yml]: when: os_images_name_suffix is None (might be shorter)

File: tasks/promote.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+- name: Ensure old images are retired
+  command: "{{ os_images_venv }}/bin/openstack image set {{ promotion_name }} --name {{ promotion_name }}.{{ date_suffix }}"
+  vars:
+    suffix: "{{ os_images_name_suffix }}"
```
Comment (mnasiadka) [inline tasks/promote.yml]: is this used?

File: tasks/promote.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    date_suffix: "{{ ansible_date_time.date }}"
+    suffix_length: "{{ os_images_name_suffix|length }}"
+    promotion_name: "{{ item.name[:-(suffix_length|int)] }}"
```
Comment (mnasiadka) [inline tasks/promote.yml]: you could use "{{ item.name[:-(os_images_name_suffix | length | int)] }}"

**Review by mnasiadka — COMMENTED** [review]
and generally Jinja2 variables and filters should have spaces before and after - should we have some linting in that repo?

File: tasks/promote.yml (hunk 3 above)
Comment (g0rgamesh) [reply tasks/promote.yml]: Unfortunately this does not work as intended:
```fatal: [localhost]: FAILED! => {"msg": "The conditional check 'os_images_name_suffix is None' failed. The error was: No test named 'None' found.\n\nThe error appears to be in '/home/cloud-user/smslab-config/ansible/roles/stackhpc.os-images/tasks/promote.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- name: Check if image suffix is provided\n  ^ here\n"}```

Comment (mnasiadka) [reply tasks/promote.yml]: oh well, then leave length, but add spaces - os_images_name_suffix | length == 0

File: tasks/promote.yml (hunk 4 above)
Comment (g0rgamesh) [reply tasks/promote.yml]: It's not, deleted

**Review by mnasiadka — APPROVED** [review]

**Pull request merged**